services/gundeck/src/Gundeck/Push/Websocket.hs

```diff
@@ -98,8 +98,10 @@ logBadCannons (uri, (err, prcs)) = do
     forM_ prcs $ \prc ->
         Log.warn $ logPresence prc
             ~~ Log.field "created_at" (ms $ createdAt prc)
-            ~~ Log.msg (val "WebSocket presence unreachable: " +++
-                        show (uri, resource prc, show err))
+            ~~ Log.field "cannon_uri" (show uri)
+            ~~ Log.field "resource_target" (show $ resource prc)
+            ~~ Log.field "http_exception" (show $ show err)  -- (double-show to kill newlines)
```

Comment:
How does the double show work?

When I try it out it just gets worse:

`> show "test\ntwo\nthree"
"\"test\\ntwo\\nthree\""
`

Reply:
worse is better!  :-)

the important thing here is to avoid printing actual newlines to stdout in the middle of a log message.  `\\\\\\\n` is fine.

with some luck there will be a better fix soon: https://gitlab.com/twittner/tinylog/merge_requests/3

Reply:
Ahh, gotcha. Until then perhaps `squashLog = intercalate " | " . lines` would make this look less like it's just a typo